### ISSUE TITLE
IDE regression: Flickering of resolution items

### ISSUE THREAD

**Issue by MikaelMayer** [dafny-lang/dafny#5605]
### Dafny version

latest-nightly

### Code to produce this issue

```dafny
Long Dafny code with a lot of resolution issues
```

### Command to run and resulting output

```code
N/A
```

### What happened?

I don't know for how long but this behavior seems to be recent. While attempting a big refactoring of a big Dafny file, I fix one resolution issue and before I have the time to click on the next, all resolution issues disappear for 4 seconds, and then reappear.

This is annoying because before, I was able to fix those items one every other second, so it doubles my waiting time now.
Since it's resolution, I can't switch to "verification on save" to avoid that behavior.

I realized that before, when resolution was working and was taking 20 long seconds, I really appreciated that all the errors went away immediately.
So I don't have a clear opinion on what to do. Looks like a tradeoff

### What type of operating system are you experiencing the problem on?

Windows

**Comment by keyboardDrummer** [comment]
Indeed, when you have to wait long before resolution diagnostics have been computed, both clearing them after an edit until they've been recomputed, or not updating them until they've been recomputed, are not ideal.

If resolution emitted events more granularly, such as once per resolved module, then all the modules for which resolution hits the cache would immediately emit their event and we could send the resolution diagnostics that were unaffected by the change. That way, after an edit, only resolution diagnostics that might be affected by the edit would disappear, until resolution of those modules finishes again.
